apps/remote_control/lib/lexical/remote_control/commands/rename.ex

```diff
@@ -0,0 +1,131 @@
+defmodule Lexical.RemoteControl.Commands.Rename do
+  # We are unable to accurately determine when the rename process finishes,
+  # because after the rename, we will receive a series of events like
+  # `DidChange`, `DidSave`, etc., which will trigger expensive actions like compiling the entire project.
+  # Therefore, we need this module to tell us if lexical is currently in the process of renaming.
+
+  alias Lexical.RemoteControl.Api.Messages
+  require Logger
+  import Messages
+
+  defmodule State do
+    defstruct uri_with_expected_operation: %{}, on_update_progess: nil, on_complete: nil
+
+    def new(uri_with_expected_operation, progress_functions) do
```

Comment:
each progress function should be a separate parameter.
a spec here would be nice.